src/DADynamoApp/DAEntrypoint.cs

```diff
@@ -0,0 +1,182 @@
+        {
+            get { return proxy; }
+        }
+
+        public static List<IUpdater> Updaters = new List<IUpdater>();
+
+        private void CheckIfPythonExists(string inputFolder)
+        {
+            var pyDir = "python-3.9.12-embed-amd64";
+            var pyRoot = Path.GetDirectoryName(model.PathManager.CommonDataDirectory);
+            var searchPath = Path.Combine(pyRoot, pyDir);
+            if (Directory.Exists(searchPath) && File.Exists(Path.Combine(searchPath, "python.exe")))
+            {
+                return;
+            }
+            Console.WriteLine($"<<!>> cPython not found at '{searchPath}'");
+            var inputDistributionPath = Path.Combine(inputFolder, $"{pyDir}.zip");
+            if (File.Exists(inputDistributionPath))
+            {
+                Console.WriteLine("<<!>> cPython distribution detected in input. Attempting to add it... ");
+                try
+                {
+                    ZipFile.ExtractToDirectory(inputDistributionPath, searchPath);
+                    Console.WriteLine("<<!>> cPython added.");
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"<<!>> Failed to add cPython: {ex.Message}");
+                }
+            }
+        }
+
+        public ExternalDBApplicationResult OnShutdown(ControlledApplication application)
+        {
+            AppDomain.CurrentDomain.UnhandledException -= CurrentDomain_UnhandledException;
+            AppDomain.CurrentDomain.ProcessExit -= CurrentDomain_ProcessExit;
+            AppDomain.CurrentDomain.AssemblyResolve -= DynamoPathResolver.ResolveAssembly;
+            return ExternalDBApplicationResult.Succeeded;
+        }
+
+        private static void CurrentDomain_ProcessExit(object? sender, EventArgs e)
+        {
+            Process proc = Process.GetCurrentProcess();
+            Console.WriteLine($"Dynamo exiting with Peak physical memory {proc.PeakWorkingSet64} bytes");
+            if (proc.HasExited)
+            {
+                Console.WriteLine($"Dynamo exiting with code {proc.ExitCode}");
+            }
+        }
+
+        private static void CurrentDomain_UnhandledException(object sender, UnhandledExceptionEventArgs e)
+        {
+            Console.WriteLine($"Unhandled exception: {e}");
+        }
+
+        public ExternalDBApplicationResult OnStartup(ControlledApplication application)
+        {
+            AppDomain.CurrentDomain.UnhandledException += CurrentDomain_UnhandledException;
+            AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;
+            AppDomain.CurrentDomain.AssemblyResolve += DynamoPathResolver.ResolveAssembly;
+
+            Console.WriteLine("<<!>> Starting to load D4DA");
+
+            try
+            {
+                RevitServices.Transactions.TransactionManager.SetupManager(new RevitServices.Transactions.AutomaticTransactionStrategy());
+                ElementBinder.IsEnabled = true;
```

Comment:
I am not sure element binding really makes sense in the context of design automation, it really only makes sense in interactive workflows.

I would disable it by default or potentially have a config option to turn it on instead of defaulting to on.

Reply:
I guess, if the workers are stateless and thorw away.